src/platformAccessory.ts

```diff
@@ -6,6 +6,7 @@ import { BasicServiceCreatorManager, ServiceCreatorManager } from './converters/
 import { BasicAccessory, BasicLogger, ServiceHandler } from './converters/interfaces';
 import { deviceListEntriesAreEqual, DeviceListEntry, isDeviceDefinition, isDeviceListEntry } from './z2mModels';
 import { BaseDeviceConfiguration } from './configModels';
+import {IClientPublishOptions, QoS} from 'mqtt';
```

Comment:
Formatting is not consistent, but given my comment on `getQosLevel`, I don't think we need this imports anyway.

Reply:
QoS import is mandatory because qos isn't number, but QoS type

`export interface IClientPublishOptions {
  /**
   * the QoS
   */
  qos?: QoS`

Reply:
I didn't have the import previously and I was still able to set the `qos`?

Reply:
I'm agree, but with a function who return number type, the compiler display a error.

""TS2322: Type 'number' is not assignable to type 'QoS | undefined'. "

Before, you set qos with valid number (1 or 2) declared type interface. But now, the function return number who can be outside of number declared in interface (0, 1 or 2).

Reply:
Aha, sorry, then we have to keep just the QoS type then.